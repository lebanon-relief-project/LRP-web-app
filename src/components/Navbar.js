import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import logo from "../assets/images/logo.png";
import { Routes } from "../App.js";
import colours from "../styles/Colours";

const Navbar = () => {
  return (
    <StyledNav>
      <StyledDiv>
        <LogoImage src={logo} alt="Logo" />
      </StyledDiv>

      <StyledUl>
        <StyledLi>
          <StyledLink to={Routes.HOME}>Home</StyledLink>
        </StyledLi>
        <StyledLi>
          <StyledLink to={Routes.HOME}>I’m looking for help</StyledLink>
        </StyledLi>
        <StyledLi>
          <StyledLink to={Routes.HOME}>I’d like to help</StyledLink>
        </StyledLi>
        <StyledLi>
          <StyledLink to={Routes.ABOUT}>About us</StyledLink>
        </StyledLi>
      </StyledUl>
    </StyledNav>
  );
};

const StyledNav = styled.nav`
  display: flex;
`;

const LogoImage = styled.img`
  width: 20%;
  height: auto;
`;

const StyledDiv = styled.div`
  min-width: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const StyledUl = styled.ul`
  display: flex;
  list-style-type: none;
  margin: 0;
  padding: 0;
  min-width: 50%;
`;

const StyledLi = styled.li`
  padding: 1rem;
  width: 100%;
`;

const StyledLink = styled(Link)`
  color: ${colours.black};
`;

export default Navbar;
