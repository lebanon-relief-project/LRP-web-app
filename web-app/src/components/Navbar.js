import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";
import logo from "../assets/images/logo.png";
import { Routes } from "../App.js";
import colours from "../styles/Colours";
import devices from "../styles/Devices";

const Navbar = ({ open }) => {
  return (
    <StyledNav open={open}>
      <StyledDiv>
        <LogoImage src={logo} alt="Logo" />
      </StyledDiv>

      <StyledUl>
        <StyledLi>
          <StyledLink to={Routes.HOME}>Home</StyledLink>
        </StyledLi>
        <StyledLi>
          <StyledLink to={Routes.HELP}>I’m looking for help</StyledLink>
        </StyledLi>
        <StyledLi>
          <StyledLink to={Routes.HOME}>I’d like to help</StyledLink>
        </StyledLi>
        <StyledLi>
          <StyledLink to={Routes.ABOUT}>About us</StyledLink>
        </StyledLi>
      </StyledUl>
    </StyledNav>
  );
};

const StyledNav = styled.nav`
  display: flex;

  @media (max-width: ${devices.mobile}) {
    width: 100%;
    z-index: 10;
    display: flex;
    flex-direction: column;
    justify-content: center;
    background: ${colours.white};;
    transform: ${({ open }) => (open ? "translateX(0)" : "translateX(-100%)")};
    height: 100%;
    padding: 2rem;
    position: fixed;
    transition: transform 0.3s ease-in-out;

    a {
      font-size: 2rem;
      text-transform: uppercase;
      padding: 2rem 0;
      font-weight: bold;
      letter-spacing: 0.5rem;
      text-decoration: none;
      transition: color 0.3s linear;
    }
  }
`;

const LogoImage = styled.img`
  width: 20%;
  height: auto;
`;

const StyledDiv = styled.div`
  min-width: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const StyledUl = styled.ul`
  display: flex;
  @media (max-width: ${devices.mobile}) {
    flex-direction: column;
  }
  list-style-type: none;
  margin: 0;
  padding: 0;
  min-width: 50%;
`;

const StyledLi = styled.li`
  padding: 1rem;
  width: 100%;
`;

const StyledLink = styled(Link)`
  color: ${colours.black};
`;

export default Navbar;
