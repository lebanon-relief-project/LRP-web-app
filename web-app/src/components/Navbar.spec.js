import React from "react";
import { render } from "@testing-library/react";
import Navbar from "./Navbar";
import "jest-styled-components";

jest.mock("react-router-dom", () => ({
  Link: ({ to }) => <div data-testid={`link_${to}`}></div>,
  NavLink: ({ to }) => <div data-testid={`navlink_${to}`}></div>,
}));

describe("the Navbar component", () => {
  let container;
  it("should match the snapshot when open is true", () => {
    ({ container } = renderNavbar(true));

    expect(container).toMatchSnapshot();
  });
  it("should match the snapshot when open is false", () => {
    ({ container } = renderNavbar(false));

    expect(container).toMatchSnapshot();
  });
});

const renderNavbar = (open) => {
  return render(<Navbar open={open} />);
};
