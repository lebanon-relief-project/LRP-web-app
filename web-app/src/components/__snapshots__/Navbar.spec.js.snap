// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Navbar component should match the snapshot when open is false 1`] = `
.c0 {
  position: fixed;
  min-width: 100%;
  max-height: 64px;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  background-color: white;
  z-index: 100;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  padding: 0;
  min-width: 40%;
}

.c2 {
  max-width: 100%;
  min-width: 50%;
}

@media (max-width:768px) {
  .c0 {
    padding: 1rem 0;
  }
}

@media (max-width:576px) {
  .c0 {
    width: 100%;
    max-height: 100%;
    z-index: 99;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    background: #ffffff;
    -webkit-transform: translateX(-100%);
    -ms-transform: translateX(-100%);
    transform: translateX(-100%);
    height: 100%;
    position: fixed;
    -webkit-transition: -webkit-transform 0.3s ease-in-out;
    -webkit-transition: transform 0.3s ease-in-out;
    transition: transform 0.3s ease-in-out;
  }

  .c0 a {
    font-family: Playfair Display;
    font-size: 30px;
    font-style: normal;
    font-weight: normal;
    line-height: 38px;
    text-align: left;
    -webkit-text-decoration: none;
    text-decoration: none;
    -webkit-transition: color 0.3s linear;
    transition: color 0.3s linear;
  }
}

@media (max-width:576px) {
  .c1 {
    margin-top: 104px;
    margin-left: 20px;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:576px) {
  .c2 {
    margin-bottom: 60px;
  }
}

@media (max-width:768px) {
  .c2 {
    padding: 0 0.5rem;
  }
}

@media (max-width:768px) {

}

@media (max-width:576px) {

}

@media (max-width:768px) {

}

@media (max-width:576px) {

}

<div>
  <nav
    class="c0"
  >
    <div
      data-testid="navlink_/"
    />
    <ul
      class="c1"
    >
      <li
        class="c2"
      >
        <div
          data-testid="navlink_/"
        />
      </li>
      <li
        class="c2"
      >
        <div
          data-testid="navlink_/help"
        />
      </li>
      <li
        class="c2"
      >
        <div
          data-testid="navlink_#"
        />
      </li>
    </ul>
  </nav>
</div>
`;

exports[`the Navbar component should match the snapshot when open is true 1`] = `
.c0 {
  position: fixed;
  min-width: 100%;
  max-height: 64px;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  background-color: white;
  z-index: 100;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  padding: 0;
  min-width: 40%;
}

.c2 {
  max-width: 100%;
  min-width: 50%;
}

@media (max-width:768px) {
  .c0 {
    padding: 1rem 0;
  }
}

@media (max-width:576px) {
  .c0 {
    width: 100%;
    max-height: 100%;
    z-index: 99;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    background: #ffffff;
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    height: 100%;
    position: fixed;
    -webkit-transition: -webkit-transform 0.3s ease-in-out;
    -webkit-transition: transform 0.3s ease-in-out;
    transition: transform 0.3s ease-in-out;
  }

  .c0 a {
    font-family: Playfair Display;
    font-size: 30px;
    font-style: normal;
    font-weight: normal;
    line-height: 38px;
    text-align: left;
    -webkit-text-decoration: none;
    text-decoration: none;
    -webkit-transition: color 0.3s linear;
    transition: color 0.3s linear;
  }
}

@media (max-width:576px) {
  .c1 {
    margin-top: 104px;
    margin-left: 20px;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:576px) {
  .c2 {
    margin-bottom: 60px;
  }
}

@media (max-width:768px) {
  .c2 {
    padding: 0 0.5rem;
  }
}

@media (max-width:768px) {

}

@media (max-width:576px) {

}

@media (max-width:768px) {

}

@media (max-width:576px) {

}

<div>
  <nav
    class="c0"
    open=""
  >
    <div
      data-testid="navlink_/"
    />
    <ul
      class="c1"
    >
      <li
        class="c2"
      >
        <div
          data-testid="navlink_/"
        />
      </li>
      <li
        class="c2"
      >
        <div
          data-testid="navlink_/help"
        />
      </li>
      <li
        class="c2"
      >
        <div
          data-testid="navlink_#"
        />
      </li>
    </ul>
  </nav>
</div>
`;
